lib/routes/oauth.js

```diff
@@ -68,8 +140,75 @@ module.exports = (log, config, oauthdb) => {
       handler: async function (request) {
         const sessionToken = request.auth.credentials
         return oauthdb.getScopedKeyData(sessionToken, request.payload)
-      }
+      },
     },
+    {
+      method: 'POST',
+      path: '/oauth/token',
+      options: {
+        validate: {
+          payload: TOKEN_PAYLOAD_SCHEMA
+        },
+        response: {
+          schema: Joi.object().keys({
+            access_token: validators.token.required(),
+            refresh_token: validators.token,
+            id_token: validators.assertion,
+            scope: validators.scope.required(),
+            token_type: Joi.string().valid('bearer').required(),
+            expires_in: Joi.number().max(MAX_TTL_S).required(),
+            auth_at: Joi.number(),
+            keys_jwe: validators.jwe.optional()
+          })
+        }
+      },
+      handler: async function (request) {
+        return oauthdb.postToken(request.payload)
+          .then(async (resp) => {
+
+            // TODO: is this how we are gonna get the UIDs now?
+            const tokenVerify = await oauthdb.postVerifyToken({
+              token: resp.access_token
+            });
+
+            const uid = tokenVerify.user;
+            const accountRecord = await db.account(uid);
+
```

Comment:
I'm aware this is all WIP, just noting that we only want to do the steps below if the token has "oldsync" scope. (Also I think you'll get the `notifyDeviceConnected` part for free if you actually create a placeholder device record at this point.

Reply:
yeap, thanks!